### ISSUE TITLE
Replace `t.ls` serialization with direct V8 object storage

### ISSUE THREAD

**Issue by David200197** [ezet-galaxy/-titanpl-core#3]
Replace the current `t.ls` implementation (Sled + string serialization) with direct V8 object storage using `v8::Global<Value>`. This eliminates serialization overhead and enables native JavaScript type support.

## Current Problem

The current `t.ls` API only accepts strings:

```typescript
ls.set(key: string, value: string): void
ls.get(key: string): string | null
```

This forces libraries like `super-ls` to serialize/deserialize JavaScript objects:

```
JavaScript Object → devalue.stringify() → string → Sled → disk
disk → Sled → string → devalue.parse() → rehydrate() → JavaScript Object
```

### Performance Cost

| Operation | Cost |
|-----------|------|
| `devalue.stringify()` | ~2-5x slower than JSON |
| `devalue.parse()` | ~2-5x slower than JSON |
| Rehydration (restore class prototypes) | Additional overhead per nested object |
| Sled I/O | Disk write/read |

### Functionality Loss

Even with `super-ls`, serialization cannot preserve:
- Functions
- WeakMap / WeakSet
- Symbols (partially)
- Closures
- Prototype chains require manual registration

## Proposed Solution

Store V8 object references directly in a Rust `HashMap`:

```rust
use std::collections::HashMap;
use std::sync::Mutex;
use v8::{Global, Value, Local};

lazy_static! {
    static ref MEMORY: Mutex<HashMap<String, Global<Value>>> = Mutex::new(HashMap::new());
}

fn ls_set(scope: &mut v8::HandleScope, key: String, value: Local<Value>) {
    let global = Global::new(scope, value);
    MEMORY.lock().unwrap().insert(key, global);
}

fn ls_get(scope: &mut v8::HandleScope, key: &str) -> Option<Local<Value>> {
    MEMORY.lock().unwrap()
        .get(key)
        .map(|g| Local::new(scope, g))
}
```

### New API

```typescript
// Store any JS value directly (no serialization)
t.ls.set(key: string, value: any): void

// Retrieve the exact same object (no deserialization)
t.ls.get<T = any>(key: string): T | null

// Check if key exists with valid value
t.ls.has(key: string): boolean

// Get or create pattern
t.ls.resolve<T>(key: string, resolver: () => T): T

// Delete key
t.ls.remove(key: string): void

// List all keys
t.ls.keys(): string[]

// Clear all storage
t.ls.clear(): void
```

#### `has(key)`
```typescript
// Returns true if key exists and value is not null/undefined
if (t.ls.has('cache')) {
    const cache = t.ls.get('cache');
}
```

#### `resolve(key, resolver)`
```typescript
// Get existing value OR create, store, and return new value
const cache = t.ls.resolve('app_cache', () => new Map());

// Equivalent to:
if (!t.ls.has('app_cache')) {
    t.ls.set('app_cache', new Map());
}
return t.ls.get('app_cache');
```

This pattern is essential for IoC containers and lazy initialization.

## Performance Comparison

| Operation | Current (Sled + serialize) | Proposed (v8::Global) |
|-----------|---------------------------|----------------------|
| Store object | ~1-10ms | ~microseconds |
| Retrieve object | ~1-10ms | ~microseconds |
| Map/Set support | Via super-ls (slow) | Native (instant) |
| Class instances | Requires hydration | Native (instant) |
| Functions | ❌ Not supported | ✅ Supported |

**Estimated speedup: 100-1000x for typical operations**

### 1. Native IoC Containers

```typescript
// Direct object storage - no serialization needed
class UserService {
    constructor(private db: Database) {}
    findUser(id: number) { return this.db.query(...); }
}

t.ls.set('userService', new UserService(db));

// Later (different request)
const service = t.ls.get('userService');
service.findUser(1);  // Works! Methods intact.
```

### 2. Complex Data Structures

```typescript
// Maps, Sets work natively
const cache = new Map([['key', 'value']]);
t.ls.set('cache', cache);

const retrieved = t.ls.get('cache');
retrieved.get('key');  // 'value' - It's the same Map!
retrieved.set('new', 'data');  // Modifications persist
```

### 3. Eliminates super-ls

With direct object storage, `super-ls` becomes unnecessary:
- No `devalue` serialization needed
- No `register()` for class hydration needed
- No rehydration logic needed
- Simpler, faster, more capable

## Trade-offs

| Aspect | Impact | Acceptable? |
|--------|--------|-------------|
| No disk persistence | Data lost on process restart | ✅ Yes - IoC services are recreated anyway |
| RAM limited | Can't store unlimited data | ✅ Yes - Use database for large data |
| Single process only | Not shared across instances | ✅ Yes - Use Redis for distributed state |
| GC prevention | Objects not collected until removed | ✅ Yes - Developer responsibility |

These trade-offs are acceptable for the primary use case: **sharing state between requests within the same process**.

## Migration Path

1. **Phase 1**: Add new methods (`has`, `resolve`) to current API
2. **Phase 2**: Implement v8::Global storage internally
3. **Phase 3**: Deprecate string-only restriction
4. **Phase 4**: Remove Sled dependency for `t.ls` (keep for other uses if needed)

### Thread Safety

Since Tokio handles requests concurrently but V8 is single-threaded, ensure:
- `Mutex` around the HashMap
- Or use V8's isolate-level storage if each request gets its own isolate

### Memory Management

```rust
// When removing, the Global handle is dropped
// V8's GC can then collect the JS object
fn ls_remove(key: &str) {
    MEMORY.lock().unwrap().remove(key);
}
```

## Why `register()` Is No Longer Needed

With the current serialization approach, `super-ls` requires registering classes with hydration functions:

```typescript
// Current approach - registration required
superLs.register(UserService, (data) => {
    const instance = new UserService(container.get(Database));
    Object.assign(instance, data);
    return instance;
});
```

**Why?** Because serialization loses the prototype chain:

```typescript
class User {
    name: string;
    getName() { return this.name; }
}

const user = new User();
user.name = 'John';

// Serialize
JSON.stringify(user)  // '{"name":"John"}'

// Deserialize
JSON.parse('{"name":"John"}')  // { name: 'John' }
// ❌ Not a User instance anymore!
// ❌ getName() is gone!
// ❌ Object.getPrototypeOf(result) === Object.prototype, not User.prototype
```

The `register()` + `hydrate()` pattern exists solely to reconstruct the class instance after deserialization.

**With direct V8 object storage:**

```typescript
const user = new User();
user.name = 'John';

t.ls.set('user', user);

const retrieved = t.ls.get('user');
// ✅ Same object reference
// ✅ retrieved instanceof User === true
// ✅ retrieved.getName() === 'John'
// ✅ Prototype chain intact
```

**No serialization = No prototype loss = No registration needed**

This eliminates:
- `superLs.register()`
- Hydration functions
- Class reconstruction logic
- The entire `super-ls` library

## Conclusion

Replacing Sled serialization with direct V8 object storage would:

1. **Improve performance** by 100-1000x
2. **Enable native type support** (Map, Set, classes, functions)
3. **Simplify the ecosystem** (eliminate need for super-ls)
4. **Better developer experience** (no registration, no hydration)

The trade-offs (no disk persistence, RAM limited) are acceptable for the intended use case of sharing state between requests.

---

**Related**: This change would make the `super-ls` extension obsolete, as `t.ls` would natively support everything `super-ls` currently provides through serialization workarounds.
